rust/gui-client/src-tauri/src/client/gui/os_linux.rs

```diff
@@ -1,6 +1,34 @@
-use anyhow::Result;
+use anyhow::{Context as _, Result};
 use tauri::api::notification::Notification;
 
+pub(crate) async fn set_autostart(enabled: bool) -> Result<()> {
+    let dir = dirs::config_local_dir()
+        .context("Can't compute `config_local_dir`")?
+        .join("autostart");
+    let link = dir.join("firezone-client-gui.desktop");
+    if enabled {
+        tokio::fs::create_dir_all(&dir)
+            .await
+            .context("Can't create autostart dir")?;
+        let target = std::path::Path::new("/usr/share/applications/firezone-client-gui.desktop");
+        tokio::fs::symlink(target, link)
```

Comment:
Is this idempotent?

Reply:
No, it errors if it's already enabled.
I'll switch it to deleting the link and recreating it, so if the link is out of date or something then enabling it will repair it.